packages/notifi-frontend-client/lib/NotifiFrontendClient.ts

```diff
@@ -73,4 +88,49 @@ export class NotifiFrontendClient {
 
     await Promise.all([saveAuthorizationPromise, saveRolesPromise]);
   }
+
+  async beginLoginViaTransaction({
+    walletBlockchain,
+    walletAddress,
+  }: BeginLoginProps): Promise<BeginLogInByTransactionResult> {
+    const { dappAddress } = this._configuration;
+
+    const result = await this._service.beginLogInByTransaction({
+      walletAddress: walletAddress,
+      walletBlockchain: walletBlockchain,
+      dappAddress,
+    });
+
+    const nonce = result.beginLogInByTransaction.nonce;
+
+    this.clientRandomUuid = await window.crypto.randomUUID();
```

Comment:
oooh ideally we can pull in a crypto dependency rather than depending on `window`

I know this is what happens in the current repo but it does feel wonky (this function will only be successful in a browser? i guess?)

Reply:
After research on this, we concluded via Slack DM that using `crypto` alone does not rely on the browser being open: https://developer.mozilla.org/en-US/docs/Web/API/crypto_property